custom_metrics/pwa.js

```diff
@@ -124,6 +129,7 @@ return {
   swRegistrationPropertiesInfo: Object.fromEntries(swRegistrationPropertiesInfo),
   windowEventListenersInfo: Object.fromEntries(windowEventListenersInfo),
   windowPropertiesInfo: Object.fromEntries(windowPropertiesInfo),
+  serviceWorkerInfo: Object.fromEntries(serviceWorkerInfo),
   //Experimental field: Heuristic to detect if a site has a service worker even if the 'serviceWorkers' field is empty (false positives).
-  serviceWorkerHeuristic: !isObjectKeyEmpty(serviceWorkers) || !isObjectKeyEmpty(workboxInfo) || !isObjectKeyEmpty(swEventListenersInfo) || !isObjectKeyEmpty(swMethodsInfo)
+  serviceWorkerHeuristic: !isObjectKeyEmpty(serviceWorkers) || !isObjectKeyEmpty(serviceWorkerInfo) || !isObjectKeyEmpty(workboxInfo) || !isObjectKeyEmpty(swEventListenersInfo) || !isObjectKeyEmpty(swMethodsInfo)
```

Comment:
We discussed that having a non-empty serviceWorkers object (i.e. that a ServiceWorker was registered AND the URL that was registered was downloaded) would be a strong indicator for this heuristic.

The others were weaker indicators (e.g. if registration code is commented out or never called) and maybe two or more of them should be required to avoid false positives.

You were going to include that enhancement in this PR I thought? But doesn’t look like you did. Did you change your mind here?

Reply:
I just did an update, with a function `calculateServiceWorkerHeuristic()` that returns true if either:

- `serviceWorkers` field is non empty. 
- 2 out of 4 of `serviceWorkerRegistrationInfo`, `workboxInfo`, `swEventListenersInfo`, `swMethodsInfo` are non-empty.

**Tests:**

- In [Twitter's case](https://webpagetest.org/result/210802_BiDcYP_3db14f75fc8722635c8b98cce5ee7a5c/?f=json), it returns true (Twitter has a SW).
- In [UnoTV's case](https://webpagetest.org/result/210802_AiDcNZ_e227d18859b34d1a30d05f0db5613617/?f=json), its false (UnoTV doesn't have a SW).

Reply:
Nice work!